source/agora/network/Clock.d

```diff
@@ -62,9 +62,42 @@ public alias GetNetTimeOffset = bool delegate (out long) @safe nothrow;
 public alias SetPeriodicTimer = void delegate (Duration, void delegate())
         @trusted nothrow;
 
+///
+public class MockClock : Clock
+{
+    private ulong time;
+
+    ///
+    public this(ulong time)
+    {
+        this.time = time;
+    }
+
+    ///
+    public void setTime(ulong time)
+    {
+        this.time = time;
+    }
+
+    /// returns time set by constructor
+    public override time_t networkTime () @safe nothrow { return cast(time_t)(time);}
+
+    /// returns time set by constructor
+    public override time_t localTime () @safe nothrow @nogc { return cast(time_t)(time);}
+
+    /// do nothing
+    public override void startSyncing () @safe nothrow {}
+
+    /// do nothing
+    public override void synchronize () @safe nothrow {}
+}
+
+
 /// Ditto
 public class Clock
 {
+    protected this(){}
```

Comment:
If it's only used by `MockClock` I'd prefer that you just pass the null arguments directly to the super constructor.